src/manifest.ts

```diff
@@ -1161,17 +1161,17 @@ export class Manifest {
       }
     };
 
-    // to minimize risks of race condition we attempt to lock branches used as the source of commits for the duration of the
-    // release process
+    // to minimize risks of race condition we attempt as early as possible to lock branches as used as the source of
+    // commits for the duration of the release process.
     let createdReleases: CreatedRelease[];
     const pullRequests = Object.values(pullRequestsByNumber);
+    let locked = false; // used to be sure we only take the fallback branch on locking errors, not when unlocking fails
     try {
       const lockedBranches = await this.lockPullRequestsChangesBranch(
         pullRequests
       );
+      locked = true;
       try {
-        // now that branches have been locked, ensure no new commits have been pushed since we created the release branch
-        await this.throwIfChangesBranchesRaceConditionDetected(pullRequests);
```

Comment:
Why did you take this check out?

Reply:
To avoid blocking the release creation in cases where new commits have been pushed to `next`. The command will still fail before re-aligning `next` to `main`, but at least github releases will be created for merged pull requests. So customers wouldn't be blocked from releasing a new version, but the `next` branch or release branch will likely need some intervention from a Stainless engineer to avoid git conflicts for the next PR release.

That felt like a better trade-off.